sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/TextAnalyticsActions.java

```diff
@@ -145,9 +164,20 @@ public TextAnalyticsActions setExtractKeyPhrasesActions(ExtractKeyPhrasesAction.
      * @param analyzeSentimentActions the list of {@link AnalyzeSentimentAction} to be executed.
      *
      * @return the {@link TextAnalyticsActions} object itself.
+     *
+     * @throws IllegalArgumentException if more than one {@link AnalyzeSentimentAction} action are passed in.
+     * Currently service v3.1 only accepts up to one action per type.
      */
     public TextAnalyticsActions setAnalyzeSentimentActions(AnalyzeSentimentAction... analyzeSentimentActions) {
+        validateActionsNumber(analyzeSentimentActions, AnalyzeSentimentAction.class.getName());
         this.analyzeSentimentActions = analyzeSentimentActions == null ? null : Arrays.asList(analyzeSentimentActions);
         return this;
     }
+
+    private void validateActionsNumber(Object[] actions, String actionType) {
+        if (actions != null && actions.length > 1) {
+            throw logger.logExceptionAsError(new IllegalArgumentException(String.format(
+                "Currently, the service can accept up to one %s. Duplicate actions not supported.", actionType)));
```

Comment:
One last suggestion: 
```suggestion
                "Currently, the service can accept up to one %s. Multiple actions of the same type are not supported.", actionType)));
```

Reply:
+1 to @deyaaeldeen error message.